fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registry/ToolBlockTransformationRegistries.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.AxeItem;
+import net.minecraft.item.HoeItem;
+import net.minecraft.item.ShovelItem;
+import net.minecraft.item.ToolMaterial;
+
+import net.fabricmc.fabric.api.content.registry.v1.util.Taggable2ObjectMap;
+
+public final class ToolBlockTransformationRegistries {
+	public static final Taggable2ObjectMap<Block, BlockState> TILLABLE_BLOCKS = new Taggable2ObjectMapRegistryImpl<>("tillable_block_registry", HoeAccessor::addTillable, HoeAccessor::removeTillable, HoeAccessor::getTilled);
+	public static final Taggable2ObjectMap<Block, BlockState> FLATTENABLE_BLOCKS = new Taggable2ObjectMapRegistryImpl<>("flattenable_block_registry", ShovelAccessor::addFlattenable, ShovelAccessor::removeFlattenable, ShovelAccessor::getFlattenable);
+	public static final Taggable2ObjectMap<Block, Block> STRIPPABLE_BLOCKS = new Taggable2ObjectMapRegistryImpl<>("strippable_block_registry", AxeAccessor::addStrippable, AxeAccessor::removeStrippable, AxeAccessor::getStrippable);
+
+	static class HoeAccessor extends HoeItem {
```

Comment:
I still think a mixin accessor is nicer than this, this has room to cause possible issues.

Reply:
Like what? it's impl so if mojang changes something it can very easily be converted to an Accessor mixin. imo having 3 less mixins for what can be done so simply is a good thing.

Reply:
Having less mixins isn't your goal. Also this is totally implementation detail, you should have 0 concern about binary compatibility here.

Reply:
Ah you use package private classes. Still these need changes; remove `private` modifier of those methods. This is not yet java 11, you don't have nest hosts so calling private from outer class will generate synthetics and is to be avoided,

Reply:
I still think this should be a mixin accessor. Its prob less likely to break due to not requiring the constructor. There is no harlm in having 4 mixins.

Reply:
That's actually a good point, I did notice in 1.16 one of the items changed their constructor. I'll change to an accessor.